pkg/operator/replacement_starter.go

```diff
@@ -339,3 +344,33 @@ func CreateOperatorStarter(ctx context.Context, authOperatorInput *authenticatio
 
 	return ret, nil
 }
+
+type featureGateAccessorFunc func(ctx context.Context, authOperatorInput *authenticationOperatorInput, informerFactories authenticationOperatorInformerFactories) (featuregates.FeatureGate, error)
+
+func defaultFeatureGateAccessor(ctx context.Context, authOperatorInput *authenticationOperatorInput, informerFactories authenticationOperatorInformerFactories) (featuregates.FeatureGate, error) {
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		status.VersionForOperatorFromEnv(), "0.0.1-snapshot",
+		informerFactories.operatorConfigInformer.Config().V1().ClusterVersions(),
+		informerFactories.operatorConfigInformer.Config().V1().FeatureGates(),
+		authOperatorInput.eventRecorder,
+	)
+	go featureGateAccessor.Run(ctx)
+	go informerFactories.operatorConfigInformer.Start(ctx.Done())
+
+	var featureGates featuregates.FeatureGate
+	select {
+	case <-featureGateAccessor.InitialFeatureGatesObserved():
+		featureGates, _ = featureGateAccessor.CurrentFeatureGates()
+	case <-time.After(1 * time.Minute):
+		return nil, fmt.Errorf("timed out waiting for FeatureGate detection")
+	}
+	return featureGates, nil
+}
+
+func momFeatureGateAccessor() featureGateAccessorFunc {
+	enabled := []ocpconfigv1.FeatureGateName{"ExternalOIDC"}
```

Comment:
Use `features.FeatureGateExternalOIDC` instead of hardcoding the name.

Reply:
Good idea, updated.